apps/tailwind-components/components/form/Field.vue

```diff
@@ -1,82 +1,87 @@
+} from "~/types/types";
+import type { CellValueType } from "metadata-utils/src/types";
+import {
+  InputBoolean,
+  InputCheckboxGroup,
+  InputListbox,
+  InputRadioGroup,
+  InputRef,
+  InputString,
+  InputTextArea,
+  InputPlaceHolder,
+} from "#components";
+const modelValue = defineModel<any>();
+const props = withDefaults(
+  defineProps<
+    InputProps & {
+      type: CellValueType;
+      label?: string;
+      description?: string | null;
+      required?: boolean;
+      errorMessage?: string | null;
+      refSchemaId?: string;
+      refTableId?: string;
+      refLabel?: string;
+      options?: IValueLabel[];
+    }
+  >(),
+  {
+    ...InputPropsDefaults,
+    required: false,
   }
+);
 
-  formFieldInput.value.validate(value);
+function typeToInputMap(type: CellValueType) {
+  switch (type) {
+    case "AUTO_ID":
+    case "STRING":
+    case "LONG":
+    case "EMAIL":
+    case "HYPERLINK":
+      return InputString;
+    case "CHECKBOX":
+      return InputCheckboxGroup;
+    case "RADIO":
+      return InputRadioGroup;
+    case "SELECT":
+      return InputListbox;
+    case "TEXT":
+      return InputTextArea;
+    case "REF":
+      return InputRef;
+    case "BOOL":
+      return InputBoolean;
+    default:
+      return InputPlaceHolder;
+  }
 }
+
+const inputComponent = computed(() => {
+  return typeToInputMap(props.type.toUpperCase());
+});
 </script>
 
 <template>
-  <div class="flex flex-col gap-1">
-    <div>
-      <label :for="column.id" class="capitalize text-title font-bold">{{
-        column.label
-      }}</label>
-      <span class="text-disabled text-body-sm ml-3" v-show="column.required"
-        >Required</span
-      >
-    </div>
-    <div class="text-input-description text-body-sm" v-if="column.description">
-      {{ column.description }}
-    </div>
-    <div>
-      <FormFieldInput
-        :type="column.columnType"
-        :id="column.id"
-        :label="column.label"
-        :refSchemaId="column.refSchemaId || schemaId"
-        :refTableId="column.refTableId"
-        :refLabel="column.refLabel || column.refLabelDefault"
-        :data="data"
-        :required="!!column.required"
-        :aria-invalid="hasError"
-        :aria-desribedBy="`${column.id}-input-error`"
-        @focus="touched = true"
-        @input="pristine = false"
-        @update:modelValue="$emit('update:modelValue', $event)"
-        @error="$emit('error', $event)"
-        @blur="$emit('blur')"
-        ref="formFieldInput"
-      ></FormFieldInput>
-      <div
-        v-if="hasError"
-        :id="`${column.id}-input-error`"
-        class="bg-invalid text-required p-3 font-bold flex items-center gap-1"
-      >
-        <BaseIcon name="info"></BaseIcon>{{ errors[0].message }}
-      </div>
-    </div>
-  </div>
+  <InputGroup
+    :id="id"
+    :label="label"
+    :errorMessage="errorMessage"
+    :description="description"
+    :required="required"
+  >
+    <component
+      v-model="modelValue"
```

Comment:
ARIA attributes are still needed here to link labels, descriptions, and errors to the input element. We could perhaps simplify things by passing the label in here to generate all required IDs, but I'm not sure if we want to push this to the form level.

Reply:
ok, then having the 'group' seperate doesn't allow us to do this easily. I move everything back into form/field as it was so the aria-describe by etc can be done properly.